linode/manager
# refactor: [M3-7334] - Improve, clean up, and simplify app entrypoint and render tree logic

## Description 📝
The goal of this PR is to simplify the entry point logic of Cloud Manager. Currently, the flow of `index.tsx ➡️ AuthenticationWrapper.tsx ➡️ App.tsx ➡️ MainContent.tsx ➡️ ...` is very complex and hard to follow, this ticket should help a little bit. There is a lot to clean up and improvement potential, but this PR is a solid start

## Changes  🔄
- Splashscreen simplification
- AuthenticationWrapper is now a hook
- IdentifyUser is now a hook
- Uses less redux to track app loading state
- Don't use unnecessarily complex preference toggle for Side Menu state, use preferences hooks directly

## Preview 📷
> [!note]
> No UI changes are expected

## How to test 🧪

### Verification steps 
- Verify Cloud Manager loads like normal
- Log out and verify you can log in without any issues
- Verify that your remain logged in when you refresh
- Verify Lish still works

## As an Author I have considered 🤔
